Gifski/ResizableDimensions.swift

```diff
@@ -0,0 +1,142 @@
+import struct CoreGraphics.CGSize
+import struct CoreGraphics.CGFloat
+
+enum DimensionsType: String, Equatable, CaseIterable {
+	case pixels
+	case percent
+}
+
+struct Dimensions: Equatable {
+	let type: DimensionsType
+	let value: CGSize
+}
+
+final class ResizableDimensions {
+	/// Minimum scaling, 1.0 being the original size
+	let minimumScale: CGFloat
+
+	/// Maximum scaling, 1.0 being the original size
+	let maximumScale: CGFloat
+
+	private(set) var currentDimensions: Dimensions
+	private let originalDimensions: Dimensions
+	private var currentScale: CGFloat
+
+	init(dimensions: Dimensions, minimumScale: CGFloat? = nil, maximumScale: CGFloat? = nil) {
+		self.originalDimensions = dimensions
+		self.currentDimensions = dimensions
+		self.minimumScale = minimumScale ?? 0.01
+		self.maximumScale = maximumScale ?? 1.0
+		self.currentScale = 1.0
+	}
+
+	private func copy() -> ResizableDimensions {
```

Comment:
Would be good to use a protocol for this. https://stackoverflow.com/a/35915186/64949

Reply:
Good idea 👍